sdks/python/apache_beam/utils/processes_test.py

```diff
@@ -103,5 +104,124 @@ def test_method_forwarding_windows(self, *unused_mocks):
         other_arg=True)
 
 
+class TestErrorHandlingCheckCall(unittest.TestCase):
+  @classmethod
+  def setup_class(cls):
+    cls.mock_get_patcher = mock.patch(\
+      'apache_beam.utils.processes.subprocess.check_call')
+    cls.mock_get = cls.mock_get_patcher.start()
+
+  @classmethod
+  def teardown_class(cls):
+    cls.mock_get_patcher.stop()
+
+  def test_oserror_check_call(self):
```

Comment:
These tests fail for me when run with 

```
python ./setup.py test -s apache_beam.utils.processes_test.TestErrorHandlingCheckOutput
```
```
======================================================================
ERROR: test_oserror_check_output_message (apache_beam.utils.processes_test.TestErrorHandlingCheckOutput)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/home/projects/beam/beam/beam/sdks/python/apache_beam/utils/processes_test.py", line 162, in test_oserror_check_output_message
    self.mock_get.side_effect = OSError()
AttributeError: 'TestErrorHandlingCheckOutput' object has no attribute 'mock_get'
```
I think the reason is that `setup_class` is never called.
But the tests pass when run inside tox suite:  `tox -e py27`.
Looking further to understand why.

Reply:
I tested and got the same result, can't understand why though. Did you find anything so far?

Reply:
I might be wrong, but I find that unittest expects the method to be `setUpClass` (https://docs.python.org/3/library/unittest.html#unittest.TestCase.setUpClass). Perhaps tox knows to run `setup_class`, but other way of running the test will miss the `setup_class` method because it's not what unittest expects?

Reply:
`setup_class` method is called by nose: https://github.com/nose-devs/nose/blob/7c26ad1e6b7d308cafa328ad34736d34028c122a/nose/suite.py#L134. tox suites executes all tests using nose, however when we run individual tests we run plain `unittest.runner`.

Reply:
Looks like `s/setup_class/setUpClass` is a quick fix. I'll send a PR. If I run into additional issues, we can consider rolling this back until a better fix is in place.

Reply:
Sent https://github.com/apache/beam/pull/8337.

Reply:
Sounds good to me.